pkg/resource/deploy/step_generator.go

```diff
@@ -421,23 +424,57 @@ func (sg *stepGenerator) inheritedChildAlias(
 		aliasName)
 }
 
+// configureGoal mutates the passed goal to inherit the correct ResourceOptions from its parent.
+func (sg *stepGenerator) configureGoal(goal *resource.Goal) result.Result {
+	originalParent := goal.Parent
+
+	if originalParent != "" {
+		parentGoal, ok := sg.resourceGoals[originalParent]
+		if !ok {
+			// Check if originalParent is an Alias
+			parentURN, ok := sg.aliases[originalParent]
```

Comment:
This won't handle multiple levels of aliasing (alias of alias of alias).
We need a loop here. Maybe this should be:

```go
if originalParent != "" {
  // Resolve alias until unaliased URN is found.
  parentURN := originalParent
  for {
    dest, ok := sg.aliases[parentURN]
    if !ok {  break } // parentURN is not an alias
    parentURN = dest
  }

  parentGoal, ok := sg.resourceGoals[parentURN]
  if !ok {
    return result.Errorf(...)
  }
}
```

(I wonder if StepGenerator already has a thing for resolving URNs to unaliased destinations.)

Reply:
Looking at @Frassle's other comment, this may be moot.